hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/SubmissionManager.java

```diff
@@ -154,6 +157,28 @@ public void submit(@NonNull final TransactionBody txBody, @NonNull final Bytes s
             final var success = transactionPool.submitApplicationTransaction(payload);
             if (success) {
                 submittedTxns.add(txId);
+
+                if (txBody.hasAtomicBatch()
```

Comment:
Don't we want this code right after the check for the parent `txId` (lines 148-151)? I.e. we want to check inner txn IDs _before_ submitting the batch txn to the transaction pool on line 157

Reply:
Ah I see what you mean, it's checked [here](https://github.com/hiero-ledger/hiero-consensus-node/blob/e6c0f3507ecf97e41d7e5a72229b13bfcf4e80bb/hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestChecker.java#L279-L279) for both parent and inner transactions 👍